tests/JobDefinitionFactoryTest.php

```diff
@@ -250,6 +261,258 @@ private function createJobDefinitionsWithConfiguration(array $jobData, array $co
-        return $factory->createFromJob($component, $job, $storageApiClient);
+        return $factory->createFromJob($component, $job, $this->getStorageApiClientMock($storageApiClient));
+    }
+
+    private function getStorageApiClientBranchMock(Client $branchClient): ClientWrapper
+    {
+        $clientWrapperMock = $this->createMock(ClientWrapper::class);
+        $clientWrapperMock->expects(self::never())->method('getBasicClient');
+        $clientWrapperMock->method('hasBranch')->willReturn(true);
+        $clientWrapperMock->method('getBranchId')->willReturn('my-branch');
+        $clientWrapperMock->method('getBranchClient')->willReturn($branchClient);
+        return $clientWrapperMock;
+    }
+
+    private function createJobDefinitionsWithBranchConfiguration(array $jobData, array $configuration): array
+    {
+        $encryptor = $this->createMock(ObjectEncryptor::class);
+        $encryptor->method('decrypt')->with($configuration)->willReturn($configuration);
+
+        $encryptorFactory = $this->createMock(ObjectEncryptorFactory::class);
+        $encryptorFactory->method('getEncryptor')->willReturn($encryptor);
+
+        $component = new Component([
+            'id' => 'my-component',
+            'data' => [
+                'definition' => [
+                    'type' => 'dockerhub',
+                    'uri' => 'keboola/docker-demo',
+                ],
+            ],
+        ]);
+
+        $job = new Job($encryptorFactory, $jobData);
+
+        $storageApiClient = $this->createMock(BranchAwareClient::class);
+        $storageApiClient->method('apiGet')
+            ->with('components/my-component/configs/my-config')
+            ->willReturn($configuration);
+
+        $factory = new JobDefinitionFactory();
+        return $factory->createFromJob($component, $job, $this->getStorageApiClientBranchMock($storageApiClient));
+    }
+
+    public function testCreateJobDefinitionWithBranchConfigId(): void
+    {
+        $configuration = [
+            'id' => 'my-config',
+            'version' => '1',
+            'state' => [],
+            'rows' => [],
+            'configuration' => [
+                'runtime' => [
+                    'foo' => 'bar',
+                ],
+            ],
+        ];
+
+        $jobData = [
+            'status' => JobFactory::STATUS_CREATED,
+            'projectId' => 'my-project',
+            'componentId' => 'my-component',
+            'branchId' => 'my-branch',
+            'configId' => 'my-config',
+        ];
+
+        $jobDefinitions = $this->createJobDefinitionsWithBranchConfiguration($jobData, $configuration);
+
+        self::assertCount(1, $jobDefinitions);
+        $jobDefinition = $jobDefinitions[0];
+
+        self::assertSame($jobData['configId'], $jobDefinition->getConfigId());
+        self::assertSame($jobData['componentId'], $jobDefinition->getComponentId());
+        self::assertSame('bar', $jobDefinition->getConfiguration()['runtime']['foo'] ?? null);
```

Comment:
K comu je tu ten `?? null`?

Reply:
No, to je taková Pepovina :) Aby ti to v testech nenapsalo undefined index `foo`, ale `bar` is not equal to `null`. Dám to pryč, tak mě to mate.